src/deluge/gui/views/automation_view.cpp

```diff
@@ -686,8 +686,15 @@ void AutomationView::renderRow(ModelStackWithAutoParam* modelStackWithParam, RGB
 
 	for (int32_t xDisplay = 0; xDisplay < kDisplayWidth; xDisplay++) {
 
-		uint32_t squareStart = getMiddlePosFromSquare(xDisplay, lengthToDisplay, xScroll, xZoom);
-		int32_t knobPos = getParameterKnobPos(modelStackWithParam, squareStart) + kKnobPosOffset;
+		int32_t knobPos = 0;
+
+		if (isAutomated) {
+			knobPos = getAverageSquareKnobPosition(modelStackWithParam, xDisplay, lengthToDisplay, xScroll, xZoom);
+		}
+		else {
+			uint32_t squareStart = getPosFromSquare(xDisplay, xScroll, xZoom);
+			knobPos = getParameterKnobPos(modelStackWithParam, squareStart) + kKnobPosOffset;
+		}
 
 		RGB& pixel = image[xDisplay];
```

Comment:
Can we render the min/max knob position in the range with like the note tail color, to give some indication that there is more data lurking below the surface?

Reply:
> Can we render the min/max knob position in the range with like the note tail color, to give some indication that there is more data lurking below the surface?

Can I address this in a follow up PR? I started digging into this but the way I want to show lower data is a somewhat complex undertaking and I don't want to accidentally break this PR or break automation view because it's going to require me playing around with the nodes a bit.